### PR TITLE
Remove monitoring component

### PR DESCRIPTION
**Pull request by shorim**
**Description**

Changes proposed in this pull request:

- Remove resources folder for monitoring component
- Remove CRDs for monitoring component
- fast-integration tests:
   - remove monitoring tests
   - remove unused audit-log tests
   - remove eventingMonitoringTest
- Adjust documentation
- Add migration guide and a cleanup script for: 
  - Deleting resources of the monitoring component (except for CRDs to avoid removing the CRDs which could be used in case the tutorial https://github.com/kyma-project/examples/tree/main/prometheus was followed)
  - Deleting the existing dashboards (configmaps) from `eventing`, `istio`, `ory`, `serverless` components
- Remove grafana dashboards (configmaps) from following components:
   - eventing 
   - ory
- set `prometheus.io/scrape` annotation to `false` for services in `ory` and `api-gateway` components, as they are not exposing custom metrics

**Related issue(s)**

#18258

### CODE REVIEW

**Review by NHingerl — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by NHingerl — DISMISSED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by mmitoraj — COMMENTED** [review]
I think we should also update: /docs/03-tutorials/_sidebar.md. PTAL.

**Review by jeremyharisch — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]

File: docs/04-operation-guides/operations/README.md [hunk 1]
```diff
@@ -12,12 +12,8 @@ This is a complete list of Kyma operation guides:
 * [Uninstall Kyma or components](07-uninstall-kyma.md)
 * [Back up Kyma](10-backup-kyma.md)
 * [Pass an access token in a request header](ac-01-pass-access-token-in-request-header.md)
-* [Grafana Dashboards for Kyma Eventing](evnt-01-eventing-dashboards.md)
-* [Kyma Eventing Metrics](evnt-02-eventing-metrics.md)
```
Comment (a-thaler) [inline docs/04-operation-guides/operations/README.md:16]: I assume the metrics itself should be still documented, only the dashboards will be moved outside

File: docs/04-operation-guides/troubleshooting/README.md [hunk 2]
```diff
@@ -42,8 +42,7 @@ See the full list of Kyma troubleshooting guides:
 
 ### Observability
 
-- [Prometheus Istio Server restarting or in crashback loop](./observability/obsv-01-troubleshoot-prometheus-istio-server-crash-oom.md)
-- [Component doesn't work after successful installation](./observability/obsv-02-troubleshoot-trace-backend-shows-few-traces.md)
+- [Trace backend shows fewer traces than you would like to see](./observability/obsv-02-troubleshoot-trace-backend-shows-few-traces.md)
```
Comment (a-thaler) [inline docs/04-operation-guides/troubleshooting/README.md:45]: Please remove it, already created a replacement: https://github.com/kyma-project/telemetry-manager/pull/470

File: docs/04-operation-guides/troubleshooting/observability/_sidebar.md [hunk 3]
```diff
@@ -2,5 +2,4 @@
 * [Operation Guides](../../README.md)
 * [Troubleshooting](../README.md)
 * [Observability](README.md)
-  * [Prometheus Istio Server restarting or in crashback loop](obsv-01-troubleshoot-prometheus-istio-server-crash-oom.md)
-  * [Component doesn't work after successful installation](obsv-02-troubleshoot-trace-backend-shows-few-traces.md)
+  * [Trace backend shows fewer traces than you would like to see](obsv-02-troubleshoot-trace-backend-shows-few-traces.md)
```
Comment (a-thaler) [inline docs/04-operation-guides/troubleshooting/observability/_sidebar.md:5]: Same as above, please delete

File: docs/assets/2.19-2.20-cleanup-monitoring.sh [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+kubectl delete -n kyma-system servicemonitors.monitoring.coreos.com monitoring-prometheus-istio-server-server --ignore-not-found
+kubectl delete -n kyma-system servicemonitors.monitoring.coreos.com monitoring-prometheus-node-exporter --ignore-not-found
+kubectl delete -n kyma-system virtualservices.networking.istio.io monitoring-grafana --ignore-not-found
+kubectl delete -n kyma-system persistentvolumeclaims prometheus-monitoring-prometheus-db-prometheus-monitoring-prometheus-0 --ignore-not-found
+kubectl delete -n kyma-system configmaps istio-control-plane-grafana-dashboard --ignore-not-found
+kubectl delete -n kyma-system configmaps istio-mesh-grafana-dashboard --ignore-not-found
+kubectl delete -n kyma-system configmaps istio-performance-grafana-dashboard --ignore-not-found
+kubectl delete -n kyma-system configmaps istio-service-grafana-dashboard --ignore-not-found
+kubectl delete -n kyma-system configmaps istio-workload-grafana-dashboard --ignore-not-found
+kubectl delete -n kyma-system configmaps function-metrics-dashboard --ignore-not-found
```
Comment (a-thaler) [inline docs/assets/2.19-2.20-cleanup-monitoring.sh:120]: There is a service 'monitoring-kubelet' in kube-system generated by the prometheus-operator, will that be deleted by the operator?

The pv for prometheus needs to get deleted explicitly. Not sure how to do that best via script.
What is with the cluster-wide resource like Clusterrole?

**Review by NHingerl — DISMISSED** [review]

**Review by a-thaler — DISMISSED** [review]

**Review by grischperl — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by jeremyharisch — APPROVED** [review]

**Review by triffer — APPROVED** [review]

**Pull request merged**